testing/integration-neo4j/src/test/java/com/neueda/jetbrains/plugin/graphdb/test/integration/neo4j/tests/cypher/editor/ArgumentHintTest.java

```diff
@@ -0,0 +1,121 @@
+package com.neueda.jetbrains.plugin.graphdb.test.integration.neo4j.tests.cypher.editor;
+
+import com.intellij.codeInsight.hint.ParameterInfoComponent;
+import com.intellij.lang.parameterInfo.CreateParameterInfoContext;
+import com.intellij.lang.parameterInfo.ParameterInfoUIContextEx;
+import com.intellij.testFramework.utils.parameterInfo.MockCreateParameterInfoContext;
+import com.intellij.testFramework.utils.parameterInfo.MockUpdateParameterInfoContext;
+import com.neueda.jetbrains.plugin.graphdb.language.cypher.editor.CypherParameterInfoHandler;
+import com.neueda.jetbrains.plugin.graphdb.language.cypher.references.CypherInvocation;
+import com.neueda.jetbrains.plugin.graphdb.test.integration.neo4j.util.base.BaseIntegrationTest;
+
+public class ArgumentHintTest extends BaseIntegrationTest {
+
+    private CypherParameterInfoHandler parameterInfoHandler;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        parameterInfoHandler = new CypherParameterInfoHandler();
+    }
+
+
+    public void testSpecialAll() {
+        doTest("RETURN ALL (x <caret>IN nodes(p) WHERE x.age > 30)",
+                "<html>(<b>variable IN list WHERE predicate</b>)</html>");
+    }
+
+    public void testSpecialAny() {
+        doTest("RETURN ANY (x IN a.array WHERE x = 'one'<caret>)",
+                "<html>(<b>variable IN list WHERE predicate</b>)</html>");
+    }
+
+    public void testSpecialExists() {
+        doTest("RETURN EXISTS(<caret>true)",
+                "<html>(<b>pattern</b>)</html>");
+    }
+
+    public void testSpecialNone() {
+        doTest("RETURN NONE (<caret>x IN nodes(p) WHERE x.age = 25)",
+                "<html>(<b>variable in list WHERE predicate</b>)</html>");
+    }
+
+    public void testSpecialSingle() {
+        doTest("RETURN SINGLE (<caret>var IN nodes(p) WHERE var.eyes = 'blue')",
+                "<html>(<b>variable in list WHERE predicate</b>)</html>");
+    }
+
+    public void testSpecialFilter() {
+        doTest("RETURN filter(<caret>x IN a.array WHERE size(x)= 3)",
+                "<html>(<b>variable IN list WHERE predicate</b>)</html>");
+    }
+
+    public void testSpecialExtract() {
+        doTest("RETURN extract(<caret>n IN nodes(p)| n.age) AS extracted",
+                "<html>(<b>variable IN list | expression</b>)</html>");
+    }
+
+    public void testSpecialReduce() {
+        doTest("RETURN reduce(<caret>totalAge = 0, n IN nodes(p)| totalAge + n.age) ",
+                "<html>(<b>accumulator = initial</b>, variable IN list | expression)</html>");
+    }
+
+    public void testBuiltIn() {
+        doTest("return toFloat(<caret>\"12\")",
+                "<html>(<b>expression</b>)</html>");
+    }
+
+    public void testProcedure() {
```

Comment:
This is not procedure. U do not have tests for:
- Procedure
- User defined function